### PR TITLE
[android-zoom] Add MTY_Zoom and MTY_Cursor modules

### PR DESCRIPTION
**Pull request by Namaneo**
* `MTY_Zoom` module: This module provides out-of-the-box coordinates conversion and cursor positioning, based of on the `MTY_EVENT_MOTION` and `MTY_EVENT_SCALE` events' data. It supports multiple modes:

    * `MTY_INPUT_MODE_TOUCHSCREEN`: In this mode, the cursor position is directly transformed from the absolute screen coordinates to the zoomed viewport, and the image is never moved.
    * `MTY_INPUT_MODE_TRACKPAD`: In this mode, the cursor position is computed relatively from motion events, and the image is moved in case the cursor hits the borders of the zoomed viewport.
    * Relative: This special mode is made for when the input data are relative and the cursor position must not be converted.

* `MTY_Cursor` module: This helper module allows to create, setup and draw a cursor. This is basically a wrapper around `MTY_WindowDrawQuad`.

Requires:
* https://github.com/chrisd1100/libmatoya/pull/58
* https://github.com/chrisd1100/libmatoya/pull/59
* https://github.com/chrisd1100/libmatoya/pull/62

### CODE REVIEW

File: src/cursor.c [hunk 1]
```diff
@@ -0,0 +1,95 @@
+#include "matoya.h"
+
+struct MTY_Cursor {
```
Comment (RonaldH) [inline src/cursor.c:3]:
```suggestion
#include "matoya.h"

#include <stdbool.h>
#include <stdint.h>

struct MTY_Cursor {
```
etc.

File: src/cursor.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+	MTY_RenderDesc desc = {0};
+
+	desc.format = MTY_COLOR_FORMAT_RGBA;
+	desc.filter = MTY_FILTER_LINEAR;
+	desc.cropWidth = ctx->width;
+	desc.cropHeight = ctx->height;
+	desc.imageWidth = ctx->width;
+	desc.imageHeight = ctx->width;
+	desc.aspectRatio = (float)ctx->width / (float)ctx->height;
```
Comment (RonaldH) [inline src/cursor.c:74]:
```suggestion
	desc.aspectRatio = (float) ctx->width / (float) ctx->height;
```

File: src/cursor.c [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	desc.blend = true;
+
+	desc.type = MTY_POSITION_FIXED;
+	desc.scale = ctx->scale;
+	desc.imageX = (int32_t)(ctx->x - (ctx->hotX * desc.scale));
+	desc.imageY = (int32_t)(ctx->y - (ctx->hotY * desc.scale));
```
Comment (RonaldH) [inline src/cursor.c:80]:
```suggestion
	desc.imageX = (int32_t) (ctx->x - (ctx->hotX * desc.scale));
	desc.imageY = (int32_t) (ctx->y - (ctx->hotY * desc.scale));
```

File: src/cursor.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+	MTY_WindowDrawQuad(ctx->app, window, ctx->image, &desc);
+}
+
+void MTY_CursorDestroy(MTY_Cursor **ctx)
+{
+	if (!(*ctx))
```
Comment (RonaldH) [inline src/cursor.c:87]:
```suggestion
	if (!ctx || !*ctx)
```

Comment (RonaldH) [reply src/cursor.c:87]: Adjust the rest per the rest of matoya standards.

File: src/zoom.c [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+	ctx->scale_image  *= scaleFactor;
+	
+	if (ctx->scale_screen < ctx->scale_screen_min) {
+		ctx->scale_screen = ctx->scale_screen_min;
+		ctx->scale_image  = ctx->scale_image_min;
+		scaleFactor = 1;
+	}
+
+	if (ctx->scale_screen > ctx->scale_screen_max) {
+		ctx->scale_screen = ctx->scale_screen_max;
```
Comment (RonaldH) [inline src/zoom.c:159]:
```suggestion
		scaleFactor = 1;

	} else if (ctx->scale_screen > ctx->scale_screen_max) {
		ctx->scale_screen = ctx->scale_screen_max;
```

**Comment by Namaneo** [comment]
Moved to:
* https://github.com/parsec-cloud/libmatoya/pull/9

**Review by danek819 — COMMENTED** [review]
# ```

```

**Pull request closed without merging**